package/src/components/Message/Message.tsx

```diff
@@ -691,200 +674,179 @@ const MessageWithContext = <
+      title: t('Delete Message'),
+      titleStyle: { color: accent_red },
+    };
+
+    const editMessage: MessageActionListItemMainProps = {
+      action: () => {
+        if (!StreamCache.getInstance().currentNetworkState) {
+          toast.show(t('Something went wrong'), 2000);
+        } else {
+          setOverlay('none');
+          if (handleEdit) {
+            handleEdit(message);
+          }
+          handleEditMessage();
+        }
+      },
+      actionType: 'editMessage',
+      icon: <Edit pathFill={grey} />,
+      title: t('Edit Message'),
+    };
+
+    const pinMessage: MessageActionListItemMainProps = {
+      action: () => {
+        setOverlay('none');
+        if (handlePinMessage) {
+          handlePinMessage(message);
+        }
+        handleTogglePinMessage();
+      },
+      actionType: 'pinMessage',
+      icon: <Pin height={23} pathFill={grey} width={24} />,
+      title: t('Pin to Conversation'),
+    };
+
+    const unpinMessage: MessageActionListItemMainProps = {
+      action: () => {
+        setOverlay('none');
+        if (handlePinMessage) {
+          handlePinMessage(message);
+        }
+        handleTogglePinMessage();
+      },
+      actionType: 'unpinMessage',
+      icon: <Unpin pathFill={grey} />,
+      title: t('Unpin from Conversation'),
+    };
+
+    const flagMessage: MessageActionListItemMainProps = {
+      action: () => {
+        if (!StreamCache.getInstance().currentNetworkState) {
+          toast.show(t('Something went wrong'), 2000);
+        } else {
+          setOverlay('alert');
+          if (message.id) {
+            Alert.alert(
+              t('Flag Message'),
+              t(
+                'Do you want to send a copy of this message to a moderator for further investigation?',
+              ),
+              [
+                { onPress: () => setOverlay('none'), text: t('Cancel') },
+                {
+                  onPress: async () => {
+                    try {
+                      if (handleFlag) {
+                        handleFlag(message);
+                      }
+                      await client.flagMessage(message.id);
+                      Alert.alert(
+                        t('Message flagged'),
+                        t('The message has been reported to a moderator.'),
+                        [
+                          {
+                            onPress: () => setOverlay('none'),
+                            text: t('Dismiss'),
+                          },
+                        ],
+                      );
+                    } catch (err) {
+                      Alert.alert(
+                        t('Something went wrong'),
+                        t("The operation couldn't be completed."),
+                        [
+                          {
+                            onPress: () => setOverlay('none'),
+                            text: t('Dismiss'),
+                          },
```

Comment:
The nesting here is quite deep, maybe we could refactor this a bit for readability? Like, just separate things out in different functions, or something. not easy to catch what the context is anymore when you get to the next couple of properties and see what the parents are